aps2scala/dump-scala.cc

```diff
@@ -1914,7 +1994,9 @@ void dump_scala_Declaration(Declaration decl,ostream& oss)
       Type fty = function_decl_type(decl);
       Declaration rdecl = first_Declaration(function_type_return_values(fty));
       Block b = function_decl_body(decl);
-      dump_function_prototype(name,fty,oss);
+      Declaration mdecl = get_enclosing_some_class_decl(decl);
+      std::set<Declaration> visisted;
+      dump_function_prototype(name,fty, check_override_decl(some_class_decl_result_type(mdecl), decl, visisted), oss);
```

Comment:
To me this looks like you might be adding overrides to functions in class declarations.  That seems fishy to me.  But it should cause many problem if it's a problem.  Does basic.sala get created correctly?
Also: what if the function is at the top-level? I'd expect this code to crash.  It doesn't crash?